### PR TITLE
build base image for multiple php versions

### PR DESCRIPTION
**Pull request by brettmc**
Allow building for multiple php versions by making Dockerfile php version an argument, and relaxing extension location (by using find instead of hard-coded location).
Updating the github workflow to build and tag base image for 7.4,8.0,8.1 (thus dropping "latest" from further updates).
Updating Makefile to accept `PHP_VERSION` for local testing against other versions.
Relax a couple of dev dependencies so they can install for 7.4->8.1.
Adding xdebug example config to .env.dist, and allow configurating xdebug via environment.
Adding a couple of whitespace-related php-cs-fixer rules.

### CODE REVIEW

File: .github/workflows/publish-otel-php-base-docker-image.yml
Comment (brettmc) [inline .github/workflows/publish-otel-php-base-docker-image.yml:38]: Using github action cache - this speeds up builds substantially (grpc!) but as a side-effect we might not get updated dependencies. Is it worth the trade-off?

File: .env.dist
Comment (tidal) [inline .env.dist:3]: Could you please remove this line?

File: README.md
Comment (tidal) [inline README.md:147]: Could you phrase this differently? The github actions are (luckily) not important at all, they merely excute our QA checks, which could be done by any other CI system.

File: .github/workflows/publish-otel-php-base-docker-image.yml
Comment (tidal) [reply .github/workflows/publish-otel-php-base-docker-image.yml:38]: It Looks like there is no way to configure a TTL or invalidate the cache. Maybe add some extra key to the cache-key to allow for invalidating when needed?

**Review by tidal — CHANGES_REQUESTED** [review]

File: docker/Dockerfile
Comment (tidal) [inline docker/Dockerfile:18]: +1 for using find, that's what I wanted to use in the first place, but I guess I messed smth up, that's why it did not work.
However if we go for a more abstracted way, we could also simply strip the debug signals in all extensions, so any .so file in the extension directory. That should not hurt if there is nothing to strip, but can probably lower the size of other extensions as well (maybe we need others in the future).

File: .github/workflows/publish-otel-php-base-docker-image.yml
Comment (brettmc) [reply .github/workflows/publish-otel-php-base-docker-image.yml:38]: The easiest way I can see is to add an extra variable to the key, that we can change (eg CACHE_VERSION) - but, that just sounds like troubleshooting and extra work for maintainers.
Given that the build takes around 20 minutes, and only runs against `main`, I think we will have less surprises in the future without cache.

Comment (tidal) [reply .github/workflows/publish-otel-php-base-docker-image.yml]: Yap, I agree. Actually the image build doesn't even need to be run after each merge, since it's not using the PHP code anyway, so once a week should be enough, no?

File: README.md
Comment (brettmc) [inline README.md:146]: "support officially supported PHP versions" per https://www.php.net/supported-versions.php ??

File: .github/workflows/publish-otel-php-base-docker-image.yml
Comment (brettmc) [reply .github/workflows/publish-otel-php-base-docker-image.yml]: Ooh, that's a good idea. It could run to schedule, and we have the option of manually running it as required.

File: README.md
Comment (tidal) [reply README.md:146]: Yap, however it would be nice if you replace "per" with "according to". I'm not trying to nit-pick, but despite the loanword "per" being used like that in English, this is not what the Latin word "per" means, so using it in that way often creates confusion for non native English speakers.

**Review by tidal — APPROVED** [review]

**Pull request merged**
